ogr/abstract.py

```diff
@@ -885,7 +885,15 @@ def add_user(self, user: str, access_level: AccessLevel) -> None:
         :param access_level: AccessLevel enum
         :return: None
         """
-        raise NotImplementedError
+        raise NotImplementedError()
+
+    def request_access(self) -> None:
```

Comment:
Does it make sense to request a different role?

At least, please, document the role we are requesting.

Reply:
I did consider this.

`project.accessrequests.create()` does not take any arguments. We cannot request a specific role, User while approving will have to specify a role.

Reply:
OK, can you please add this note to the docstring?

And it would be nice to be able to provide some description/note that will be shown to the fork owner. But it looks like we can't set anything else via GitLab API (https://docs.gitlab.com/ce/api/access_requests.html).

Reply:
Added in the docstring.